libs/wordpress-plugin/plugin/trunk/block/shared/block-loader/load-remote-block.tsx

```diff
@@ -59,9 +60,40 @@ export const parseBlockSource = (
    */
   const exports: Record<string, UnknownBlock> = {};
   const module = { exports };
+
+  /**
+   * If the block has been portalled into an iFrame, we need to overwrite its globals, because:
+   * - within a block, globals will refer to those in the parent window (where the block is constructed), not iFrame it is rendered in
+   * - styling libraries which work by appending styles to the head via globals are going to add styles to the parent document
+   *   - e.g. webpack style-loader (for imported CSS files), or emotion
+   * - styles are therefore not present in the iFrame unless we overwrite the globals (they'd be in the parent's <head>)
+   * - event listeners in some libraries are also likely to be affected
+   *
+   * We therefore check if the Gutenberg iFrame is present, and if so overwrite the value of key globals to those of the iFrame.
+   */
+  const iframe = getContainingIframe();
+  const documentBlockRenderedInside = iframe?.contentDocument ?? document;
+  const windowBlockRenderedInside = iframe?.contentWindow ?? window;
+
   // eslint-disable-next-line no-new-func
-  const func = new Function("require", "module", "exports", source);
-  func(requires, module, exports);
+  const func = new Function(
+    "require",
+    "module",
+    "exports",
+    "document",
+    "window",
+    "globalThis",
+    source,
+  );
+
+  func(
+    requires,
+    module,
+    exports,
+    documentBlockRenderedInside,
+    windowBlockRenderedInside,
+    windowBlockRenderedInside,
```

Comment:
Yep – unfortunately there are a bunch of ways via which the global this can be accessed. in practice blocks will be using `window` or `globalThis` (`globalThis` was introduced as a way of accessing the global object across environments, including Node).

It seems that at least Chrome sets `globalThis` and `window` separately, rather than setting `globalThis` to point to `window`, so both need to be overwritten

i.e. Chrome isn't doing this
```ts
const window = MyWindow;
const globalThis = window;
```
but this
```ts
const window = MyWindow;
const globalThis = MyWindow;
```
or at least that's what the practical effect suggests, which is that both need to be overwritten or some code breaks